dds/DCPS/DiscoveryBase.h

```diff
@@ -1161,7 +1161,7 @@ namespace OpenDDS {
           }
 
           // change this if 'writer_active' (above) changes
-          if (call_writer && !call_reader && !is_opendds(reader)) {
+          if (call_writer && !call_reader) {
```

Comment:
The call to `association_complete` below could result in two calls to association_complete for the same peer, which the writer is not expecting.